jwst/ami/leastsqnrm.py

```diff
@@ -800,6 +800,43 @@ def populate_symmamparray(amps, n=7):
 
     return arr
 
+def t3_amplitudes(amps, n=7):
+    """
+    Populate the triple-product amplitude array
+    (NOT closure amplitudes)
+
+    Parameters
+    ----------
+    amps: 1D float array
+        fringe visibility between each pair of holes
+
+    n: integer
+        number of holes
+
+    Returns
+    -------
+    cpamps: 1D float array
+        triple product amplitude array
+    """
+
+    arr = populate_symmamparray(amps, n=n)
+
+    cpamps = np.zeros(int(comb(n, 3)))
+
+    nn = 0
+    for kk in range(n - 2):
+        for ii in range(n - kk - 2):
+            for jj in range(n - kk - ii - 2):
+                cpamps[nn + jj] = (
+                    arr[kk, ii + kk + 1]
+                    * arr[ii + kk + 1, jj + ii + kk + 2]
+                    * arr[jj + ii + kk + 2, kk]
+                )
```

Comment:
Is there any way to avoid a triple `for` loop here using matrix multiplication and/or numpy array comprehension?

I haven't done any runtime profiling of this code, and I would only improve this if it was causing long runtimes.  And I suppose there are only 35 baselines for JWST, so probably this isn't too big an issue for now

Reply:
There's probably a way, but I'm hesitant to touch it right now since I'm not the original author of this function and as you've said I don't think it's likely to be a big contributor to the long runtimes. I'm definitely open to reassessing in a future build though.